### PR TITLE
[WEB] Show airqloud monitor count according to user's location

### PR DESCRIPTION
**Pull request by OchiengPaul442**
#### Summary of Changes (What does this PR do?)
NB: Added localstorage and geolocation functionality to track the user's location while using the web app.
- Implemented localStorage to store the user's country, latitude, and longitude
- Added the getUserCountryFromStorage function to retrieve country from localStorage or call getUserGeolocation
- Added getUserGeolocation function to get the user's current location and country using Geolocation API
- Created a distance function to calculate the distance between current and stored locations so that if the user's location changes it's automatically updated when they exceed 100 km of their location.
- Used an if statement to check if the country stored in local storage is included in the list of AirQlouds available and if not then it displays Uganda AirQloud by default.

- Also Updated some content on the API page.

#### NB: Added Jest Test Library for Unit Testing

- This commit adds the Jest Test library and its required dependencies to the website, enabling us to carry out unit tests. This will help us ensure that our code is functioning as intended and improve the overall quality of our website. 
#### Refactored Code to Resolve Issues
- This commit includes refactoring various files to resolve issues that were identified during testing.

#### Status of maturity (all need to be checked before merging):
- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] Included unit test for the Geolocate component

- [Jira_card_number]() https://airqoteam.atlassian.net/browse/WEB-406

#### Screenshots (optional)
image
image
image
image

### CODE REVIEW

File: website/frontend/src/components/Footer.js [hunk 1]
```diff
@@ -94,6 +96,83 @@ const Footer = () => {
...
+      } else {
+        getUserGeolocation();
+      }
+    };
+
+    // Function to fetch users country based on their IP address
+    const getUserCountry = async (latitude, longitude) => {
+      try {
+        const response = await axios.get(
+          `https://api.bigdatacloud.net/data/reverse-geocode-client?latitude=${latitude}&longitude=${longitude}&localityLanguage=en`
```
Comment (Baalmart) [inline website/frontend/src/components/Footer.js:113]: thanks @OchiengPaul442 ,can we have this url stored somewhere else? and just referenced here as a variable? In our codebase, do we have somewhere we keep such constants like urls? Thanks again!

**Review by Baalmart — COMMENTED** [review]
thanks @OchiengPaul442 , this is good work. I have just left one small comment.
Additionally, now is a good time to start accompanying your new implementation with unit tests.

**Comment by OchiengPaul442** [comment]
Thanks, @Martin Bbaale ***@***.***>, for your valuable feedback. Yes,
there is a URL file in the project, and will work on implementing this
change. and I forgot 😅 about the unit test, but I will implement that on
the next commit.

Thanks.

**Review by Baalmart — APPROVED** [review]

File: website/frontend/config/urls.js [hunk 2]
```diff
@@ -62,3 +62,6 @@ export const EVENTS_URL = `${WEBSITE_BASE_URL}/event/`;
 
 // African Cities url
 export const CITIES_URL = `${WEBSITE_BASE_URL}/african_city/`;
+
+// Location tracking url
+export const LOCATIONS_TRACKING_URL = `${WEBSITE_BASE_URL}/https://api.bigdatacloud.net/data/reverse-geocode-client`;
```
Comment (Baalmart) [inline website/frontend/config/urls.js:67]: please review this url, something is off here.

**Review by Baalmart — CHANGES_REQUESTED** [review]

**Comment by Baalmart** [comment]
@OchiengPaul442  , review why we are having the video in the landing page and yet it is not supposed to be in different PR?

**Comment by OchiengPaul442** [comment]
Thanks @Baalmart, working on it.

**Pull request closed without merging**